Clean up some uses of `new` and `delete` in drag and drop code inside WebViewImpl

whsieh opened the pull request:
#### 1941a4dbfa8e13933d56b5f23aabadad6f6a5f0d
<pre>
Clean up some uses of `new` and `delete` in drag and drop code inside WebViewImpl
<a href="https://bugs.webkit.org/show_bug.cgi?id=242113">https://bugs.webkit.org/show_bug.cgi?id=242113</a>
Work towards rdar://96074318

Reviewed by Tim Horton.

Remove a few direct calls to `new` and `delete` in `WebViewImpl`; this was used to accumulate both
`DragData` and a list of file paths when handling drops on macOS by allocating these non-refcounted
objects on the heap. However, a cleaner way to achieve this is to use `WTF::Box`, which wraps these
objects in a ref-counted struct, and is deleted after all of the file promise receivers&apos; completion
handlers are done.

* Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
(WebKit::WebViewImpl::performDragOperation):

Canonical link: <a href="https://commits.webkit.org/251960@main">https://commits.webkit.org/251960@main</a>
</pre>

hortont424 commented on Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
```diff
@@ -4343,31 +4343,24 @@ static NSDragOperation kit(std::optional<WebCore::DragOperation> dragOperation)
-
-            RetainPtr<NSOperationQueue> queue = adoptNS([NSOperationQueue new]);
-            [item receivePromisedFilesAtDestination:dropDestination options:options operationQueue:queue.get() reader:^(NSURL *fileURL, NSError *errorOrNil) {
+        [draggingInfo enumerateDraggingItemsWithOptions:0 forView:m_view.getAutoreleased() classes:@[[NSFilePromiseReceiver class]] searchOptions:@{ } usingBlock:[&](NSDraggingItem *draggingItem, NSInteger idx, BOOL *stop) {
```
while you're here, slap some spaces inside that array literal

whsieh replied:
Ah, I actually left that alone because of:
```
ERROR: Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:4357:  Extra space in capture list.  [whitespace/brackets] [4]
Total errors found: 1 in 1 files
```
...but I'll fix it here, and ignore the style checker false positive.

Thanks for the review!

whsieh committed:
Clean up some uses of `new` and `delete` in drag and drop code inside WebViewImpl
https://bugs.webkit.org/show_bug.cgi?id=242113
Work towards rdar://96074318

Reviewed by Tim Horton.

Remove a few direct calls to `new` and `delete` in `WebViewImpl`; this was used to accumulate both
`DragData` and a list of file paths when handling drops on macOS by allocating these non-refcounted
objects on the heap. However, a cleaner way to achieve this is to use `WTF::Box`, which wraps these
objects in a ref-counted struct, and is deleted after all of the file promise receivers' completion
handlers are done.

* Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
(WebKit::WebViewImpl::performDragOperation):

Canonical link: https://commits.webkit.org/251960@main
